src/Symfony/Component/DependencyInjection/Compiler/FactoryReturnTypePass.php

```diff
@@ -51,12 +51,12 @@ public function process(ContainerBuilder $container)
     private function updateDefinition(ContainerBuilder $container, $id, Definition $definition, array $resolveClassPassChanges, array $previous = array())
     {
         // circular reference
-        if (isset($previous[$id])) {
+        if (isset($previous[$lcId = strtolower($id)])) {
```

Comment:
I would move ``$lcId = strtolower($id)`` on its own statement. It would be more readable